src/test/java/org/jboss/xavier/integrations/route/XmlRoutes_RestReportTest.java

```diff
@@ -520,6 +530,266 @@ public void xmlRouteBuilder_RestReportIdWorkloadInventory_IdParamGiven_ShouldCal
+        workloadInventoryReportModels.add(wir1);
+
+        WorkloadInventoryReportModel wir2 = new WorkloadInventoryReportModel();
+        wir2.setProvider("ProviderB");
+        wir2.setDatacenter("DatacenterA");
+        wir2.setCluster("ClusterB");
+        wir2.setVmName("VmNameA");
+        workloadInventoryReportModels.add(wir2);
+
+        WorkloadInventoryReportModel wir3 = new WorkloadInventoryReportModel();
+        wir3.setProvider("ProviderA");
+        wir3.setDatacenter("DatacenterC");
+        wir3.setCluster("ClusterA");
+        wir3.setVmName("VmNameC");
+        workloadInventoryReportModels.add(wir3);
+
+        analysisService.addWorkloadInventoryReportModels(workloadInventoryReportModels, analysisModel.getId());
+
+
+        Map<String, Object> variables = new HashMap<>();
+        variables.put("id", analysisModel.getId());
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set(TestUtil.HEADER_RH_IDENTITY, TestUtil.getBase64RHIdentity());
+        HttpEntity<String> entity = new HttpEntity<>(null, headers);
+
+        //When
+        camelContext.start();
+        TestUtil.startUsernameRoutes(camelContext);
+        camelContext.startRoute("to-sortBean");
+        camelContext.startRoute("to-workloadInventoryFilterBean");
+        camelContext.startRoute("filtered-workload-inventory-report-get-details-as-csv");
+        camelContext.startRoute("workload-inventory-report-model-to-csv");
+
+        ResponseEntity<String> response = restTemplate.exchange(camel_context + "report/{id}/workload-inventory/filtered-csv", HttpMethod.GET, entity, String.class, variables);
+
+        //Then
+        SortBean sortBean = new SortBean(null, true);
+        WorkloadInventoryFilterBean filterBean = new WorkloadInventoryFilterBean();
+
+        verify(workloadInventoryReportService).findByAnalysisOwnerAndAnalysisId(analysisModel.getOwner(), analysisModel.getId(), sortBean, filterBean);
+        Assert.assertTrue(response.getHeaders().get("Content-Type").contains("text/csv"));
+        Assert.assertTrue(response.getHeaders().get("Content-Disposition").contains("attachment;filename=workloadInventory_1.csv"));
+        Assert.assertNull(response.getHeaders().get("whatever"));
+        assertThat(response).isNotNull();
+        assertThat(response.getBody()).contains("Provider,Datacenter,Cluster,VM name,OS type,Operating system description,Disk space,Memory,CPU cores,Workload,Effort,Recommended targets,Flags IMS,Product,Version,HostName");
+
+//        Expected CSV result
+//        ProviderA | DatacenterC | ClusterA | VmNameC
+//        ProviderB | DatacenterA | ClusterB | VmNameA
+//        ProviderB | DatacenterB | ClusterB | VmNameB
+        assertThat(response.getBody()).contains("ProviderA,DatacenterC,ClusterA,VmNameC");
+        assertThat(response.getBody()).contains("ProviderB,DatacenterA,ClusterB,VmNameA");
+        assertThat(response.getBody()).contains("ProviderB,DatacenterB,ClusterB,VmNameB");
```

Comment:
If I'm not wrong you are checking that the body containes those 3 texts, but not the order of them , as every assert will return True no matter the order they have.
Is that what you wanted ?

Reply:
@jonathanvila thanks for noticing that... I've added a new commit where I'm splitting the response's body into rows and then check the order.